packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -239,6 +240,8 @@ class RebaseQueue {
 function addMovedMarkEffect(mark: Mark, effect: MarkEffect): Mark {
 	if (isMoveIn(mark) && isMoveOut(effect)) {
 		return { ...mark, type: "Insert" };
+	} else if (isRename(mark) && isMoveOut(effect)) {
+		return { ...effect, count: mark.count, idOverride: mark.idOverride };
```

Comment:
We can do this in a separate PR, but we can probably adjust the type for moved marks so that it's clear that `effect` is a type that extends `DetachFields`.

Reply:
Created https://dev.azure.com/fluidframework/internal/_workitems/edit/9387